objectherkenning_openbare_ruimte/on_edge/detection_pipeline/components/model_result.py

```diff
@@ -0,0 +1,314 @@
+        annotation_str = self._get_annotation_string_from_boxes(self.boxes[target_idxs])
+        annotation_path = os.path.join(
+            image_detection_path, f"{image_file_name.stem}.txt"
+        )
+        with open(annotation_path, "w") as f:
+            f.write(annotation_str)
+
+    def _get_annotation_string_from_boxes(self) -> str:
+        annotation_lines = []
+
+        for box in self.boxes:
+            cls = int(box.cls.squeeze())
+            conf = float(box.conf.squeeze())
+            tracking_id = int(box.id.squeeze()) if box.is_track else -1
+            yolo_box_str = " ".join([f"{x:.6f}" for x in box.xywhn.squeeze()])
+            annotation_lines.append(f"{cls} {yolo_box_str} {conf:.6f} {tracking_id}")
+
+        return "\n".join(annotation_lines)
+
+    def _yolo_result_summary(self) -> List[str]:
+        """Returns a readable summary of the results.
+
+        Returns
+        -------
+        Dict
+            Readable summary of objects detected and compute used.
+        """
+        obj_classes, obj_counts = np.unique(self.result.boxes.cls, return_counts=True)
+        obj_str = "Detected: {"
+        for obj_cls, obj_count in zip(obj_classes, obj_counts):
+            obj_str = obj_str + f"{self.result.names[obj_cls]}: {obj_count}, "
+        if len(obj_classes):
+            obj_str = obj_str[0:-2]
+        obj_str = obj_str + "}"
+
+        speed_str = "Compute: {"
+        for key, value in self.result.speed.items():
+            speed_str = speed_str + f"{key}: {value:.2f}ms, "
+        speed_str = speed_str[0:-2] + "}"
+
+        return [obj_str, speed_str]
+
+    def yolo_annotation_to_bounds(
+        self, yolo_annotation: str, img_shape: Tuple[int, int]
+    ) -> Tuple[int, int, int, int]:
+        """
+        Convert YOLO annotation with normalized values to absolute bounds.
+
+        Parameters
+        ----------
+        yolo_annotation : str
+            YOLO annotation string in the format:
+            "<class_id> <x_center_norm> <y_center_norm> <w_norm> <h_norm>".
+        img_shape : Tuple[int, int]
+            Image dimensions as tuple (width, height)
+
+        Returns
+        -------
+        tuple
+            A tuple (x_min, y_min, x_max, y_max).
+        """
+        _, x_center_norm, y_center_norm, w_norm, h_norm = map(
+            float, yolo_annotation.split()[0:5]
+        )
+
+        x_center_abs = x_center_norm * img_shape[0]
+        y_center_abs = y_center_norm * img_shape[1]
+        w_abs = w_norm * img_shape[0]
+        h_abs = h_norm * img_shape[1]
+
+        x_min = int(x_center_abs - (w_abs / 2))
+        y_min = int(y_center_abs - (h_abs / 2))
+        x_max = int(x_center_abs + (w_abs / 2))
+        y_max = int(y_center_abs + (h_abs / 2))
+
+        return x_min, y_min, x_max, y_max
+
+    def blur_inside_boxes(
+        self,
+        boxes: Union[List[Tuple[float, float, float, float]], npt.NDArray[np.float_]],
+        blur_kernel_size: int = 165,
```

Comment:
These arguments (kernel_size and padding) should perhaps be config params. Not really urgent though.

Reply:
Same for arguments in the following methods.